streaming/src/main/scala/org/apache/spark/streaming/dstream/ReceiverInputDStream.scala

```diff
@@ -54,6 +55,14 @@ abstract class ReceiverInputDStream[T: ClassTag](@transient ssc_ : StreamingCont
 
   def stop() {}
 
+  def setPreferredLocation(_preferredLocation: String){
+    if (null != _preferredLocation){
+      this._preferredLocation = Some(_preferredLocation)
+    }
```

Comment:
This can be simplified as `this._preferredLocation = Option(_preferredLocation)`.